lib/fizzy/parser_expr.cpp

```diff
@@ -100,10 +134,16 @@ parser_result<Code> parse_expr(const uint8_t* pos, const uint8_t* end, const Mod
         auto& frame = control_stack.top();
         const auto& metrics = metrics_table[opcode];
 
-        if (frame.stack_height < metrics.stack_height_required && !frame.unreachable)
-            throw validation_error{"stack underflow"};
+        if (!frame.unreachable)
+        {
+            if (frame.stack_height - metrics.stack_height_required < frame.parent_stack_height)
+                throw validation_error{"stack underflow"};
+
+            frame.stack_height += metrics.stack_height_change;
```

Comment:
I understand that you do it only for reachable instructions, so that unreachable code doesnt't influence `code.max_stack_height`.

But it doesn't look correct to update `frame.stack_height` here only when not unreachable, und update it always in `end` instruction (line 325, parent frame stack shouldn't be increased after passing unreachable block then) and also in `update_caller_frame` (line 91, unreachable `call` shouldn't update stack)

Reply:
Maybe it would be better to update `code.max_stack_height` only when reachable and `frame.stack_height` always?

Reply:
It was like that before, so I will revert the change.